src/contracts/exchange/NiftyswapExchange20.sol

```diff
@@ -160,6 +184,26 @@ contract NiftyswapExchange20 is ReentrancyGuard, ERC1155MintBurn, INiftyswapExch
     return price; // Will add 1 if rounding error
   }
 
+  /**
+   * @dev Pricing function used for converting Tokens to currency token (including royalty fee)
+   * @param _tokenId            Id ot token being sold
+   * @param _assetBoughtAmount  Amount of Tokens being bought.
+   * @param _assetSoldReserve   Amount of currency tokens in exchange reserves.
+   * @param _assetBoughtReserve Amount of Tokens (output type) in exchange reserves.
+   * @return price Amount of currency tokens to send to Niftyswap.
+   */
+  function getBuyPriceWithRoyalty(
```

Comment:
if royalties are enabled on the pool, then they must be paid in each txn right..? Im wondering if its necessary to have getBuyPriceWithRoyalty? (/sell), or if we can just stick to the same previous getBuyPrice method, and return the value with or without royalties depending on the config? This makes it simpler when querying price data from UX to keep the same method name, and let it output the price to buy as required

Reply:
if anything, we could include `getBuyPriceWithoutRoyalty` if someone wants to query the price if the royalty wasnt included.. but default `getBuyPrice` should be the same method to place the buy order

Reply:
What people will use is `getPrice_tokenToCurrency()` and `getPrice_currencyToToken()`, which return the total amount of currency user will receive for the former and the minimum amount of currency the user needs to send for the latter.

Reply:
ok thanks for clarifying